### PR TITLE
Apache ShenYu (incubating) plugin support grpc,sofarpc,motan,tars

### PR DESCRIPTION
**Pull request by hutaishi**
Apache ShenYu (incubating) plugin Enhancement: support grpc,sofarpc,motan

- [x] Add a test case for the new plugin, refer to [the doc](https://github.com/apache/skywalking-java/blob/main/docs/en/setup/service-agent/java-agent/Plugin-test.md)
- [x] Update the documentation to include this new feature.
- [x] If it's UI related, attach the screenshots below.

### grpc
image

image

### sofarpc
image

image

### motan rpc
image
image

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Release Notes.
 * Change the dateformat of logs to `yyyy-MM-dd HH:mm:ss.SSS`(was `yyyy-MM-dd HH:mm:ss:SSS`).
 * Fix NPE in elasticsearch plugin.
 * Grpc plugin support trace client async generic call(without grpc stubs), support Method type: `UNARY`、`SERVER_STREAMING`.
+* Enhance Apache ShenYu (incubating) plugin: support trace `grpc`,`sofarpc`,`motan`,`tars` rpc proxy.
```
Comment (wu-sheng) [inline CHANGES.md:17]: tars? I am concerned about this, we don't have tars plugin for the service's RPC. This new feature seems a little misguided.

Also, supported-list.md should be updated too.

Comment (hutaishi) [reply CHANGES.md:17]: This can only be supported to the gateway level, not to the service provider, because there is no skywalking tars plugin.

Comment (wu-sheng) [reply CHANGES.md:17]: That is sensitive about how we are going to write doc to be clear about this.

Comment (hutaishi) [reply CHANGES.md:17]: I remove tars from the md document

Comment (wu-sheng) [reply CHANGES.md]: Oh, I didn't mean change log, you could make it back.

I mean here https://github.com/apache/skywalking-java/blob/main/docs/en/setup/service-agent/java-agent/Supported-list.md. Add supported protocols(including tars) in here clearly.

Comment (hutaishi) [reply CHANGES.md:17]: Understood, thanks

File: docs/en/setup/service-agent/java-agent/Supported-list.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ metrics based on the tracing data.
   * JRE HttpURLConnection (Optional²)
 * HTTP Gateway
   * [Spring Cloud Gateway](https://spring.io/projects/spring-cloud-gateway) 2.0.2.RELEASE -> 3.x (Optional²)
-  * [Apache ShenYu](https://shenyu.apache.org) 2.4.x (Optional²)
+  * [Apache ShenYu](https://shenyu.apache.org) (Rich protocol support: `Http`,`Spring Cloud`,`Grpc`,`Dubbo`,`SOFARPC`,`Motan`,`tars`) 2.4.x (Optional²)
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/Supported-list.md:34]:
```suggestion
  * [Apache ShenYu](https://shenyu.apache.org) (Rich protocol support: `Http`,`Spring Cloud`,`gRPC`,`Dubbo`,`SOFARPC`,`Motan`,`Tars`) 2.4.x (Optional²)
```

File: apm-sniffer/optional-plugins/shenyu-2.4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/shenyu/v24x/GrpcPluginInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * shenyu GrpcPlugin Interceptor.
+ */
+public class GrpcPluginInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                             MethodInterceptResult result) throws Throwable {
+        CommonUtil.createLocalSpan(allArguments, "GrpcPlugin");
```
Comment (wu-sheng) [inline apm-sniffer/optional-plugins/shenyu-2.4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/shenyu/v24x/GrpcPluginInterceptor.java:36]:
```suggestion
        CommonUtil.createLocalSpan(allArguments, "gRPCPlugin");
```

gRPC is the right one, unless you have to put `Grpc` on the class name.

**Comment by wu-sheng** [comment]
Generally good, just one suggestion.

**Comment by hutaishi** [comment]
> Generally good, just one suggestion.

File: apm-sniffer/optional-plugins/shenyu-2.4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/shenyu/v24x/GrpcPluginInterceptor.java (hunk 3 above)
Comment (hutaishi) [reply apm-sniffer/optional-plugins/shenyu-2.4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/shenyu/v24x/GrpcPluginInterceptor.java:36]: OK.thinks

**Comment by hutaishi** [comment]
> Generally good, just one suggestion.

Ok,thinks. Will think about better practice later

File: apm-sniffer/optional-plugins/shenyu-2.4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/shenyu/v24x/GrpcPluginInterceptor.java (hunk 3 above)
Comment (hutaishi) [reply apm-sniffer/optional-plugins/shenyu-2.4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/shenyu/v24x/GrpcPluginInterceptor.java]: Modified

**Review by wu-sheng — APPROVED** [review]
Once tests pass, I am good.

**Pull request merged**